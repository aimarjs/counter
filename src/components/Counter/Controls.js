import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import Button from '../Button';

const Controls = ({
  onAdd, onReset, onRemove, value,
}) => (
  <Wrapper>
    <Button title="Add" clickHandler={onAdd} />
    <Button title="Reset" clickHandler={onReset} disable={value === 0} />
    <Button title="Remove" clickHandler={onRemove} disable={value === 0} />
  </Wrapper>
);

const Wrapper = styled.div`
  margin: 0 auto;
  max-width: 300px;
`;

Controls.propTypes = {
  onAdd: PropTypes.func.isRequired,
  onReset: PropTypes.func.isRequired,
  onRemove: PropTypes.func.isRequired,
  value: PropTypes.number.isRequired,
};

export default Controls;
