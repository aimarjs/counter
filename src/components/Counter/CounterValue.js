import React from 'react';
import styled from 'styled-components';

const CounterValue = props => (
  <Value>{props.value}</Value>
);

const Value = styled.h2`
  font-size: 64px;
`;

export default CounterValue;
