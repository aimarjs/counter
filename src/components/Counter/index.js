import React from 'react';
import styled from 'styled-components';
import Button from '../Button';
import CounterValue from './CounterValue';

const Counter = props => (
  <Wrapper>
    <CounterValue value={props.value} />
    <Controls>
      <Button title="Add" clickHandler={props.onAdd} />
      <Button title="Reset" clickHandler={props.onReset} disable={props.value === 0} />
      <Button title="Remove" clickHandler={props.onRemove} disable={props.value === 0} />
    </Controls>
  </Wrapper>
);

const Wrapper = styled.div`
  width: 100%
`;

const Controls = styled.div`
  margin: 0 auto;
  max-width: 300px;
`;

export default Counter;
