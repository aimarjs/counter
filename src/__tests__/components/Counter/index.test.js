import React from 'react';
import renderer from 'react-test-renderer';

import Counter from '../../../components/counter';

describe('Counter Component', () => {
  it('capturing snapshot of Counter', () => {
    const onAdd = () => {};
    const onRemove = () => {};
    const onReset = () => {};
    const value = 0;

    const renderedValue = renderer.create(
      <Counter
        value={value}
        onAdd={onAdd}
        onRemove={onRemove}
        onReset={onReset}
      />,
    ).toJSON();
    expect(renderedValue).toMatchSnapshot();
  });
  // let wrapper; let onAdd; let onRemove; let onReset;
  // beforeEach(() => {
  //   const value = 10;
  //   onAdd = jest.fn();
  //   onRemove = jest.fn();
  //   onReset = jest.fn();
  //
  //   wrapper = shallow(
  //     <Counter
  //       value={value}
  //       onAdd={onAdd}
  //       onRemove={onRemove}
  //       onReset={onReset}
  //     />,
  //   );
  //   // console.log(wrapper.find());
  //   it('should render current counter value', () => {
  //
  //     expect(wrapper.find('Add').text()).toContain('10');
  //   });
  // });
});
