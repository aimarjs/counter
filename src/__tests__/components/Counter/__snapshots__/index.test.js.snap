// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Counter Component capturing snapshot of Counter 1`] = `
<div
  className="sc-htpNat AzUzt"
>
  <h2
    className="sc-bwzfXH kMlfnI"
  >
    0
  </h2>
  <div
    className="sc-bxivhb hvkEvt"
  >
    <button
      className="sc-bdVaJa bDsUlA"
      disabled={false}
      onClick={[Function]}
      type="button"
    >
      Add
    </button>
    <button
      className="sc-bdVaJa bDsUlA"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Reset
    </button>
    <button
      className="sc-bdVaJa bDsUlA"
      disabled={true}
      onClick={[Function]}
      type="button"
    >
      Remove
    </button>
  </div>
</div>
`;
